text/0006-folio-distributed-vs-centralized-configration.md

```diff
@@ -0,0 +1,93 @@
+* Write-only configuration values are possible, like for credentials. The module can offer other operators than reading values like comparing hashes (possible in central configuration too?)
+* Modules can handle upgrade of configuration variable names or values during module upgrades more flexible
+
+Even when there are also some drawbacks on distributed configuration, it is the preferred way to configure backend modules in FOLIO.
+
+### When to use central configuration
+
+mod-settings solves the security problems of mod-configuration. It is the preferred module if configuration variables shall be stored centrally. It is not recommended to develop specialized modules for other central configuration store.
+
+Centralized configuration can only be used for either:
+
+* Non-sensitive information, that are used by many modules or are completely independent of any module. One example are locale settings.
+* Configurations that are specific to a user.
+
+While these configurations can also be stored decentralized in a module, the developer can decide where these values shall be stored.
+
+### Migration
+
+See Timing Section below.
+
+## Risks and Drawbacks
+
+Migrations from old mod-configuration can fail, and therefore tenant upgrades may fail.
+Distributed configuration requires more developer effort than central configuration.
+
+## Rationale and Alternatives
+
+Both central and distributed configurations were discussed. 
+While pure central configuration offer easier access to the complete configuration of a tenant, this is not desireable sice:
+* it is a antipattern in a microservice architecture 
+* validation of values is not possible
+* differentiating configuration data and application data is not easy: are circulation rules configuration data or application data?
+
+A pure distributed configuration has the following drawbacks:
+* for centrally used configuration values, like locale settings, a owning module has to be found. It may not be intuitive finding out, which module holds these configurations.
+
+Therefore a distributed configuration with some exceptions has been considered.
+
+## Timing
+
+Since most modules already store configuration values in a distributed way, only some cases need to be addressed.
+For locale properties and other properties still residing exclusively in mod-configuration, the access to these properties has to be moved to the module (distributed configuration, prferred) or to the mod-settings (centralized configuration, not preferred) until the Ramsons release. Therefore a mod-configuration module offering only READ and DELETE APIs will run in Ramsons and the modules still using mod-configuration have to transfer their properties to mod-settings or to a distributed configuration. Migrated configurations in mod-configuration have to be deleted. This can be done during module upgrades.
+mod-configuration will be removed in the release following the Ramsons release.
```

Comment:
Is this RFC asserting the Ramsons release? OR this is a decision made prior to this RFC?

Reply:
```suggestion
mod-configuration will be removed in the release following the Ramsons release.
In Ransoms, modules, that used mod-configuration still keep the dependency to mod-configuration. A dummy mod-configuration module can be used to reveal modules that still try to use mod-configuration.
```

Reply:
nitpick: This RFC reiterates multiple times that mod-configuration will be removed in the Ramsons release. If for some reason mod-configuration is removed in the Sunflower release, this RFC will document some timing that would not be correct. If the scope of this RFC is just to remove mod-configuration, it may be good to not add time bounds. Timing would be a implementation detail.